library/src/main/java/org/apache/apex/malhar/lib/state/spillable/SpillableStateStore.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.lib.state.spillable;
+
+import org.apache.apex.malhar.lib.state.BucketedState;
+
+import com.datatorrent.api.Component;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.Operator;
+
+public interface SpillableStateStore extends BucketedState, Component<Context.OperatorContext>,
```

Comment:
@davidyan74 
the components that we write need to work in context of operator which is why they have ```beginWindow``` and ```endWindow``` in their contract. However, all these components provide different functionalities and therefore an api with multiple implementations.

They are not interchange-able. For eg, there is a ```WindowBoundedService``` which has ```beginWindow``` and ```endWindow``` in its API but there is no practical use case where someone will use a ```WindowListener``` and set WindowBounderService as the implementation.